src/core/define.ml

```diff
@@ -6,34 +6,76 @@ type define = {
 	mutable defines_signature : string option;
 }
 
+type user_define = {
+	doc : string;
+	flags : define_parameter list;
+	source : string option;
+}
+
+let user_defines : (string, user_define) Hashtbl.t = Hashtbl.create 0
+
+let register_user_define s data =
+	Hashtbl.replace user_defines s data
+
+type define_origin =
+	| Compiler
+	| UserDefined of string option
+
+let infos d = match d with
+	| Custom(s) when (Hashtbl.mem user_defines s) ->
+		let infos = Hashtbl.find user_defines s in
+		(s, (infos.doc, infos.flags), (UserDefined infos.source))
+	| _ ->
```

Comment:
Should this really match the `Custom` case where the `mem` check fails? I'm not sure what that actually means.

Reply:
Hmm yeah you're right, I copy/pasted it from meta but for defines Custom is only for user defines.